gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2562,18 +2903,33 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
 
     if( !m_osGridMapping.empty() && nMainDimId == m_nRecordDimID )
     {
-        status =
-            nc_put_att_text(m_nLayerCDFId, nVarID, CF_GRD_MAPPING,
+        if ( !m_bLegacyCreateMode )
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation